src/deadline/maya_submitter/assets.py

```
@@ -291,6 +295,8 @@ def _get_node_attr_paths(self, expand_tokens=False):
                     attrs = []
                 attrs.append("%s.absoluteCacheName" % node)
             if attrs is not None:
+                if excluded_attrs := excluded_attrs_by_node.get(str(node), set()):
```

Why do we need a default value for the `excluded_attrs_by_node` dictionary? We are setting it above? In what scenario could this happen? Even more so, if the dictionary is empty, are we ok with the error being silenced, or should we throw an exception?

---

We are looping through all the nodes, and getting all the attributes in each of the nodes. For all these nodes except for vray settings, there won't be any excluded attributes, so it is an empty set, and it doesn't loop through it.